routers/static_filter.go

```
@@ -95,6 +96,41 @@ func fastAutoSignin(ctx *context.Context) (string, error) {
 	return res, nil
 }
 
+func InsertOrganizationTheme(ctx *context.Context, urlPath string) {
+	if urlPath == "/login" {
+		organization, err := object.GetOrganization(fmt.Sprintf("admin/built-in"))
+		if err == nil && organization != nil {
+			themeDataString, err := json.Marshal(organization.ThemeData)
+			if err == nil {
+				ctx.SetCookie("organizationTheme", string(themeDataString))
+			}
+		}
+	} else if strings.HasPrefix(urlPath, "/login/oauth/authorize") {
+		clientId := ctx.Input.Query("client_id")
```

If clientId is empty, then silently return with no error